api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -200,6 +193,10 @@ private void updateDataAccessLevel(DbUser dbUser, Agent agent) {
       addToRegisteredTierGroupIdempotent(dbUser);
       newDataAccessLevel = DataAccessLevel.REGISTERED;
 
+      // record the user as having Registered Tier membership in user_access_tier
+      // which will eventually serve as the source of truth (TODO)
+      accessTierService.addUserToRegisteredTier(dbUser);
```

Comment:
The add/remove should perhaps instead be found within the L213 conditional. This section will be invoked repeatedly with the same value. By my reading of the code that would mean that `lastUpdated()` is getting constantly modified, without the actual value changing. I'm not sure that is desired (also on a lot of unrelated interactions, it might generate traffic).

Reply:
My intention here was to follow the existing add/remove idempotent logic.  I agree that excessive changes could be bad, so I can make it only update when necessary.